arch/wasm32/pthread_arch.h

```diff
@@ -1,4 +1,4 @@
-static inline struct pthread *__pthread_self(void) { return pthread_self(); }
+static inline struct pthread *__pthread_self(void) { return NULL; }
```

Comment:
In a wasm embedding this will ~never cause a crash. Could you instead return the largest valid pointer, so it's guaranteed to crash if used?

Reply:
huh, cool idea! Kinda makes me wonder how broken everything would be if we always treat `nullptr` as `-1`...

Reply:
Totally valid to do! 😁